### PR TITLE
[FLS-60] Fix delete/register user with equal data in forum

### PR DESCRIPTION
**Pull request by dyudyunov**
**Description:** Bugfix for discussion opening for recreated user with equal username

**Youtrack:** https://youtrack.raccoongang.com/issue/FLS-60

**Testing instructions:**
1. Register on platform as student
2. Enroll to any course
3. Click view course, then click discussion
4. Enter admin page http://localhost:18000/admin/auth/user/ and delete your student account created earlier 
5. Sigh out and repeat p1-3
6. There should not be any exceptions and discussion opens correctly

**Post merge:**
- [ ] Delete working branch (if not needed anymore)

### CODE REVIEW

File: lms/lib/comment_client/user.py [hunk 1]
```diff
@@ -180,6 +185,15 @@ def retire(self, retired_username):
             metric_tags=self._metric_tags
         )
 
+    def delete(self):
+        hash = hex(int(time.time()))[2:]
```
Comment (OPersian) [inline lms/lib/comment_client/user.py]: @dyudyunov please rename the var, `hash` is a reserved name

Comment (dyudyunov) [reply lms/lib/comment_client/user.py]: thanks, I'll update it

File: lms/lib/comment_client/user.py [hunk 2]
```diff
@@ -180,6 +185,15 @@ def retire(self, retired_username):
             metric_tags=self._metric_tags
         )
 
+    def delete(self):
```
Comment (OPersian) [inline lms/lib/comment_client/user.py]: I took another look at this method, and the naming seems a bit misleading. We don't actually delete anything but run custom logic which is called before actual user removal (as far as I see). So maybe `update_username` or smth like that would be more suitable... What are your thoughts? @dyudyunov

Comment (dyudyunov) [reply lms/lib/comment_client/user.py]: Agree with You, thanks

Comment (cmltaWt0) [reply lms/lib/comment_client/user.py]: Waiting for change.

Comment (cmltaWt0) [reply lms/lib/comment_client/user.py]: @dyudyunov change method naming or add an appropriate docstring please.

File: lms/lib/comment_client/user.py [hunk 3]
```diff
@@ -180,6 +185,15 @@ def retire(self, retired_username):
...
+        hashed = hex(int(time.time()))[2:]
+        self.username = "{0}_{1}".format(self.username, hashed)
+        self.save()
+
+@receiver(pre_delete, sender=django_user)
+def user_delete_handler(sender, **kwargs):
```
Comment (cmltaWt0) [inline lms/lib/comment_client/user.py]: Here an appropriate docstring must be added.

File: lms/lib/comment_client/user.py [hunk 4]
```diff
@@ -5,6 +5,11 @@
 import models
 import utils
 
+import time
```
Comment (cmltaWt0) [inline lms/lib/comment_client/user.py]: Please rearrange imports according to the OpenEdx code style guide.
https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/style_guides/python-guidelines.html#imports-order

**Pull request merged**